specs/capella/beacon-chain.md

```diff
@@ -377,23 +276,57 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
         process_execution_payload(state, block.body.execution_payload, EXECUTION_ENGINE)  # [Modified in Capella]
     process_randao(state, block.body)
     process_eth1_data(state, block.body)
-    process_operations(state, block.body)
+    process_operations(state, block.body)  # [Modified in Capella]
     process_sync_aggregate(state, block.body.sync_aggregate)
 ```
 
+#### New `get_expected_withdrawals`
+
+```python
+def get_expected_withdrawals(state: BeaconState) -> Sequence[Withdrawal]:
+    epoch = get_current_epoch(state)
+    withdrawal_index = state.next_withdrawal_index
+    validator_index = state.latest_withdrawal_validator_index 
```

Comment:
Went through the PR, if you insist on this I can change this to have the double `%` in the logic here. It will not change formally the spec. If however you also want to start from *-1* at the fork, that could be debatable :)

Reply:
next would change the spec, no? As it would start at and utilize 0

Reply:
if it helps at all I think we will be happier in the long run if we have symmetry here with the naming, even if that implies a double modulo operation

Reply:
I'm confused as there are two or three issues here at the same time. I'm happy to add the extra ÷ here and this will start the sweep at zero as Danny mentions. This is fine by me. Naming this variable as `latest_validator_index` is good by me too. If these are the requests I'll push this in a couple of hours when I'm at the keyboard, just need nudge here to see if we're on the same page

Reply:
Added the extra % and changing the index in the end of the loop. Didn't change the variable name cause it felt odd to call it latest when it's now the current.

Reply:
I think we've been talking past eachother, I apologize.

I think that both vars should be called `next_*`

In the beginning of `get_expected_withdrawals` you would assign `validator_index = state.next_withdrawal_validator_index` and each loop you would increment (with the mod).  Then in `process_withdrawals` you would do `state.next_withdrawal_validator_index = (latest_withdrawal.validator_index + 1) % len(state.validators)`. This is the second mod I am referring to, and ensures that the withdrawal index and withdrawal_validator_index parallel eachother in usage in the loop (assign, then increment. Rather than increment then assign).

Reply:
Just like here -- https://github.com/ethereum/consensus-specs/pull/3068/files#diff-e62664a360f87fbbd16c6f0f0d11f0e47dcc4f343dc6478b6f81c3f5acd6d8cbL246

it's the next place to look on the next sweep

I'll do a quick pr to this pr. apologies for my failure to communicate clearly

Reply:
Pushed these changes, named the state counter now `next_withdrawal_validator_index`. Let me know if this works for you and I'll change our deployed code :)